### PR TITLE
Use LoggerMessage.Define for logs

### LINKED ISSUE #20: Logging review
We should do a pass over the logging statements and fix up a few things:

* Ensure every event has a Name. I don't feel a need for a numeric ID, but I'm fine either way. Names are easier to keep unique than IDs.
* Use `LoggerMessage.Define` where reasonable.

### LINKED ISSUE #66: Add CategoryName To IOperationLogger
### What should we add or change to make your life better?
Create and use `IOperationLogger<TCategoryName>`  instead of `IOperationLogger`  and make the implementations use `ILogger<TCategoryName>` insead of `ILogger`.
### Why is this important to you?
It's easier to follow logs.

### PR DESCRIPTION
**Pull request by Kahbazi**
- Added CategoryName to `IOperationLogger`
- Use LoggerMessage.Define
- Added Event Ids in a single file so it's easier to prevent using a same event id

Fixes #20 
Fixes #66

### CODE REVIEW

File: src/ReverseProxy.Core/Configuration/ProxyConfigLoader.cs [hunk 1]
```diff
@@ -83,7 +84,7 @@ private async Task ApplyAsync(ProxyConfigOptions config)
                 return;
             }
 
-            _logger.LogInformation("Applying proxy configs");
+            _logger.LogInformation(EventIds.ApplyProxyConfig, "Applying proxy configs");
```
[the remark this answers is not in the document]
Comment (Kahbazi) [reply src/ReverseProxy.Core/Configuration/ProxyConfigLoader.cs:87]: I wanted to ask, should I use `LoggerMessage.Define` for log messages that don't have any parameters?

File: src/ReverseProxy.Utilities/RecurrentActivities/RecurrentActivityManager.cs [hunk 2]
```diff
@@ -100,8 +100,7 @@ public void AddWork(string operationName, TimeSpan schedulingInterval, Func<Task
         /// <inheritdoc />
         public void AddWork(string operationName, TimeSpan schedulingInterval, Func<CancellationToken, Task> func, bool executeImmediately = false)
         {
-            var workItem = new WorkItem
-            {
+            var workItem = new WorkItem {
```
[the remark this answers is not in the document]
Comment (Kahbazi) [reply src/ReverseProxy.Utilities/RecurrentActivities/RecurrentActivityManager.cs:103]: This has been done by (Ctrl+K, Ctrl+D). Should I add `object_collection_array_initializers` to `csharp_new_line_before_open_brace` in `editor.config`?

File: src/ReverseProxy.Core/Configuration/ProxyConfigLoader.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tratcher) [reply src/ReverseProxy.Core/Configuration/ProxyConfigLoader.cs:87]: I'm not sure what the official guidance is, but the inconsistency bothers me.

@rynowak you changed the logger define pattern last, no?

File: src/ReverseProxy.Utilities/RecurrentActivities/RecurrentActivityManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tratcher) [reply src/ReverseProxy.Utilities/RecurrentActivities/RecurrentActivityManager.cs:103]: Yeah, I thought I had fixed all of those...

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**